cmd/kuberhealthy/reaper.go

```diff
@@ -224,7 +227,7 @@ func (k *KubernetesAPI) deleteFilteredCheckerPods(ctx context.Context, client *k
 		// Delete if there are more than MaxPodsThreshold checker pods with the same name in status Succeeded that were created more recently
 		// Delete if the checker pod is Failed and there are more than MaxPodsThreshold Failed checker pods of the same type which were created more recently
 		allCheckPods := getAllPodsWithCheckName(reapCheckerPods, v)
-		if len(allCheckPods) > cfg.MaxCheckPods {
+		if len(allCheckPods) > cfg.MaxCompletedCheckPods {
```

Comment:
Thinking about it more... this is probably all made more complicated by configuring both a max age for pods and a max count... maybe we should pick one.  Either we delete based on how many pods there are, or we delete after a specific age.  A specific age seems like the easiest way to do it - you wouldn't have to select and delete the oldest pods in that logic... Only the pods that are older than a timestamp.

Reply:
Not quite, `failedPodCleanupDurationDefault` or the max age for pods apply only to failed pods (pods in error state), not completed pods which is why we have them configured separately. Also, I do think using max pod count is important because if we only specify using a timestamp, checks that run every minute will be will be much more prolific compared to checks that run every hour or later. 

BUT, I do agree these configurations are complicated and there is definitely room for simplification...

Reply:
Currently the logic is 
Failed Pods:
- Keep around for certain duration or UP UNTIL count of failed pods > `maxCompletedCheckPods`

Completed Pods:
- Keep around until count of completed pods > `maxCompletedCheckPods`

I think the reason we want to keep the pod count around is because of the different check run intervals